I fixed the json test.

Now there's only one pylint test failing
> tests/api_connexion/endpoints/test_config_endpoint.py:45:8: W0201: Attribute 'client' defined outside __init__ (attribute-defined-outside-init)

I converted unittest to pytest as I remember there was a discussion to move away from unittest to pytest.
Can I make `client` a class attribute by moving it to `setup_class()`? Was there any reason to put it in `setUp()` rather than `setUpClass()` for unittest?

--- a/airflow/api_connexion/endpoints/config_endpoint.py
+++ b/airflow/api_connexion/endpoints/config_endpoint.py
@@ -31,11 +31,11 @@ def _conf_dict_to_config(conf_dict: dict) -> Config:
             ConfigSection(
                 name=section,
                 options=[
-                    ConfigOption(key=key, value=value, source=source)
-                    for key, (value, source) in parameters.items()
+                    ConfigOption(key=key, value=value)
+                    for key, value in options.items()
                 ]
             )
-            for section, parameters in conf_dict.items()
+            for section, options in conf_dict.items()
         ]
     )
     return config
@@ -43,7 +43,7 @@ def _conf_dict_to_config(conf_dict: dict) -> Config:
 
 def _option_to_text(config_option: ConfigOption) -> str:
     """Convert a single config option to text"""
-    return f'{config_option.key} = {config_option.value}  # source: {config_option.source}'
+    return f'{config_option.key} = {config_option.value}'
 
 
 def _section_to_text(config_section: ConfigSection) -> str:
@@ -72,7 +72,7 @@ def get_config() -> Response:
     }
     response_types = serializer.keys()
     return_type = request.accept_mimetypes.best_match(response_types)
-    conf_dict = conf.as_dict(display_source=True, display_sensitive=True)
+    conf_dict = conf.as_dict(display_source=False, display_sensitive=True)
     config = _conf_dict_to_config(conf_dict)
     if return_type not in serializer:
         return Response(status=406)

--- a/airflow/api_connexion/endpoints/config_endpoint.py
+++ b/airflow/api_connexion/endpoints/config_endpoint.py
@@ -21,14 +21,11 @@ from airflow.api_connexion.schemas.config_schema import Config, ConfigOption, Co
 from airflow.configuration import conf
 from airflow.settings import json
 
+LINE_SEP = '\n'  # `\n` cannot appear in f-strings
 
-def get_config() -> Response:
-    """
-    Get current configuration.
-    """
-    response_types = ['text/plain', 'application/json']
-    return_type = request.accept_mimetypes.best_match(response_types)
-    conf_dict = conf.as_dict(display_source=True, display_sensitive=True)
+
+def _conf_dict_to_config(conf_dict: dict) -> Config:
+    """Convert config dict to a Config object"""
     config = Config(
         sections=[
             ConfigSection(
@@ -41,16 +38,44 @@ def get_config() -> Response:
             for section, parameters in conf_dict.items()
         ]
     )
-    if return_type == 'text/plain':
-        config_text = '\n'.join(
-            f'[{config_section.name}]\n' +
-            ''.join(f'{config_option.key} = {config_option.value}  # source: {config_option.source}\n'
-                    for config_option in config_section.options)
-            for config_section in config.sections
-        )
-        return Response(config_text, headers={'Content-Type': return_type})
-    elif return_type == 'application/json':
-        config_text = json.dumps(config_schema.dump(config).data, indent=4)
-        return Response(config_text, headers={'Content-Type': return_type})
-    else:
+    return config
+
+
+def _option_to_text(config_option: ConfigOption) -> str:
+    """Convert a single config option to text"""
+    return f'{config_option.key} = {config_option.value}  # source: {config_option.source}'
+
+
+def _section_to_text(config_section: ConfigSection) -> str:
+    """Convert a single config section to text"""
+    return (f'[{config_section.name}]{LINE_SEP}'
+            f'{LINE_SEP.join(_option_to_text(option) for option in config_section.options)}{LINE_SEP}')
+
+
+def _config_to_text(config: Config) -> str:
+    """Convert the entire config to text"""
+    return LINE_SEP.join(_section_to_text(s) for s in config.sections)
+
+
+def _config_to_json(config: Config) -> str:
+    """Convert a Config object to a JSON formatted string"""
+    return json.dumps(config_schema.dump(config).data, indent=4)
+
+
+def get_config() -> Response:
+    """
+    Get current configuration.
+    """
+    serializer = {
+        'text/plain': _config_to_text,
+        'application/json': _config_to_json,
+    }
+    response_types = serializer.keys()
+    return_type = request.accept_mimetypes.best_match(response_types)
+    conf_dict = conf.as_dict(display_source=True, display_sensitive=True)
+    config = _conf_dict_to_config(conf_dict)
+    if return_type not in serializer:
         return Response(status=406)
+    else:
+        config_text = serializer[return_type](config)
+        return Response(config_text, headers={'Content-Type': return_type})

--- a/airflow/api_connexion/openapi/v1.yaml
+++ b/airflow/api_connexion/openapi/v1.yaml
@@ -1757,9 +1757,6 @@ components:
         value:
           type: string
           readOnly: true
-        source:
-          type: string
-          readOnly: true
 
     ConfigSection:
       type: object

--- a/airflow/api_connexion/schemas/config_schema.py
+++ b/airflow/api_connexion/schemas/config_schema.py
@@ -24,14 +24,12 @@ class ConfigOptionSchema(Schema):
     """ Config Option Schema """
     key = fields.String(required=True)
     value = fields.String(required=True)
-    source = fields.String(required=True)
 
 
 class ConfigOption(NamedTuple):
     """ Config option """
     key: str
     value: str
-    source: str
 
 
 class ConfigSectionSchema(Schema):

--- a/tests/api_connexion/endpoints/test_config_endpoint.py
+++ b/tests/api_connexion/endpoints/test_config_endpoint.py
@@ -23,11 +23,11 @@ from airflow.www import app
 
 MOCK_CONF = {
     'core': {
-        'parallelism': ('1024', 'env var'),
+        'parallelism': '1024',
     },
     'smtp': {
-        'smtp_host': ('localhost', 'default'),
-        'smtp_mail_from': ('airflow@example.com', 'airflow.cfg'),
+        'smtp_host': 'localhost',
+        'smtp_mail_from': 'airflow@example.com',
     },
 }
 
@@ -50,11 +50,11 @@ class TestGetConfig:
         assert response.status_code == 200
         expected = textwrap.dedent("""\
         [core]
-        parallelism = 1024  # source: env var
+        parallelism = 1024
 
         [smtp]
-        smtp_host = localhost  # source: default
-        smtp_mail_from = airflow@example.com  # source: airflow.cfg
+        smtp_host = localhost
+        smtp_mail_from = airflow@example.com
         """)
         assert expected == response.data.decode()
 
@@ -66,14 +66,14 @@ class TestGetConfig:
                 {
                     'name': 'core',
                     'options': [
-                        {'key': 'parallelism', 'value': '1024', 'source': 'env var'},
+                        {'key': 'parallelism', 'value': '1024'},
                     ]
                 },
                 {
                     'name': 'smtp',
                     'options': [
-                        {'key': 'smtp_host', 'value': 'localhost', 'source': 'default'},
-                        {'key': 'smtp_mail_from', 'value': 'airflow@example.com', 'source': 'airflow.cfg'},
+                        {'key': 'smtp_host', 'value': 'localhost'},
+                        {'key': 'smtp_mail_from', 'value': 'airflow@example.com'},
                     ]
                 },
             ]

--- a/tests/api_connexion/endpoints/test_config_endpoint.py
+++ b/tests/api_connexion/endpoints/test_config_endpoint.py
@@ -40,6 +40,7 @@ class TestGetConfig:
     @classmethod
     def setup_class(cls) -> None:
         cls.app = app.create_app(testing=True)  # type:ignore
+        cls.client = None
 
     def setup_method(self) -> None:
         self.client = self.app.test_client()  # type:ignore

--- a/tests/api_connexion/schemas/test_config_schema.py
+++ b/tests/api_connexion/schemas/test_config_schema.py
@@ -25,14 +25,14 @@ class TestConfigSchema:
                 ConfigSection(
                     name='sec1',
                     options=[
-                        ConfigOption(key='apache', value='airflow', source='airflow.cfg'),
-                        ConfigOption(key='hello', value='world', source='default'),
+                        ConfigOption(key='apache', value='airflow'),
+                        ConfigOption(key='hello', value='world'),
                     ]
                 ),
                 ConfigSection(
                     name='sec2',
                     options=[
-                        ConfigOption(key='foo', value='bar', source='env var'),
+                        ConfigOption(key='foo', value='bar'),
                     ]
                 ),
             ]
@@ -43,14 +43,14 @@ class TestConfigSchema:
                 {
                     'name': 'sec1',
                     'options': [
-                        {'key': 'apache', 'value': 'airflow', 'source': 'airflow.cfg'},
-                        {'key': 'hello', 'value': 'world', 'source': 'default'},
+                        {'key': 'apache', 'value': 'airflow'},
+                        {'key': 'hello', 'value': 'world'},
                     ]
                 },
                 {
                     'name': 'sec2',
                     'options': [
-                        {'key': 'foo', 'value': 'bar', 'source': 'env var'},
+                        {'key': 'foo', 'value': 'bar'},
                     ]
                 },
             ]

